src/Illuminate/Database/Eloquent/Relations/BelongsToMany.php

```diff
@@ -611,6 +611,30 @@ public function findMany($ids, $columns = ['*'])
     }
 
     /**
+     * Find a related model by its primary key or throw an exception.
+     *
+     * @param  mixed  $id
+     * @param  array  $columns
+     * @return \Illuminate\Database\Eloquent\Model|\Illuminate\Database\Eloquent\Collection
+     *
+     * @throws \Illuminate\Database\Eloquent\ModelNotFoundException
+     */
+    public function findOrFail($id, $columns = ['*'])
+    {
+        $result = $this->find($id, $columns);
+
+        if (is_array($id)) {
+            if (count($result) == count(array_unique($id))) {
```

Comment:
Well the [Eloquent Builder uses `==`](https://github.com/laravel/framework/blob/5.1/src/Illuminate/Database/Eloquent/Builder.php#L122)